### PR TITLE
chore: improve external secret reconciliation sequence

### PR DESCRIPTION
**Pull request by merlindorin**
Secret client is created only if we are going to refresh. This skip an unnecessary check/request in the case we are not going to do anything.

(hello guys, it's me again 👋 :P )

### CODE REVIEW

File: pkg/controllers/externalsecret/externalsecret_controller.go [hunk 1]
```diff
@@ -151,16 +151,6 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
-	if err != nil {
-		log.Error(err, errStoreClient)
-		conditionSynced := NewExternalSecretCondition(esv1beta1.ExternalSecretReady, v1.ConditionFalse, esv1beta1.ConditionReasonSecretSyncedError, errStoreClient)
-		SetExternalSecretCondition(&externalSecret, *conditionSynced)
-		r.recorder.Event(&externalSecret, v1.EventTypeWarning, esv1beta1.ReasonProviderClientConfig, err.Error())
-		syncCallsError.With(syncCallsMetricLabels).Inc()
-		return ctrl.Result{RequeueAfter: requeueAfter}, nil
-	}
-
 	defer func() {
```
Comment (gusfcarvalho) [inline pkg/controllers/externalsecret/externalsecret_controller.go:154]: We need to move this defer as well, or any hits on line 188 and 192 will cause a panic.

Comment (merlindorin) [reply pkg/controllers/externalsecret/externalsecret_controller.go]: 😢 didn't check, #shameOnMe

**Comment by merlindorin** [comment]
#done should be ready for milestone 0.6.0 ;)

**Comment by moolen** [comment]
damn, there's a conflict again :disappointed: can you please rebase again :pray:

**Comment by merlindorin** [comment]
@moolen #done, sry for the delay

**Comment by moolen** [comment]
Thank you again for your contributions!

**Pull request merged**